app/models/enterprise_relationship.rb

```diff
@@ -22,7 +22,8 @@ class EnterpriseRelationship < ActiveRecord::Base
   }
 
   scope :involving_enterprises, ->(enterprises) {
-    where('parent_id IN (?) OR child_id IN (?)', enterprises, enterprises)
+    enterprise_ids = enterprises.map(&:id)
+    where('parent_id IN (?) OR child_id IN (?)', enterprise_ids, enterprise_ids)
```

Comment:
With map you are forcing an extra query. I think you should use select here and make it a subquery:

```suggestion
        enterprise_ids = enterprises.select(&:id)
```